tools/lib/provision_inner.py

```diff
@@ -91,6 +93,34 @@ def configure_rabbitmq_paths() -> list[str]:
     return paths
 
 
+def is_wsl_instance() -> bool:
+    if "WSL_DISTRO_NAME" in os.environ:
+        return True
+
+    try:
+        with open("/proc/version") as file:
+            content = file.read().lower()
+            if "microsoft" in content:
+                return True
+    except FileNotFoundError:
+        pass
+
+    try:
+        result = subprocess.run(["uname", "-r"], capture_output=True, text=True, check=True)
+        if "microsoft" in result.stdout.lower():
+            return True
+    except Exception:
+        pass
```

Comment:
What expected exceptions do you intend here? It is in general not a good idea to catch all exceptions, as the code inside crashing always would be invisible with that. Is it just `subprocess.CalledProcessError`?